src/test/proxy/server_proxy.py

```diff
@@ -0,0 +1,128 @@
+from ja.user.proxy import IUserServerProxy
+from ja.common.proxy.ssh import SSHConfig, ISSHConnection
+from ja.common.message.base import Response
+from ja.common.job import Job, JobStatus, JobPriority
+from ja.user.config.add import AddCommandConfig
+from ja.user.message.query import QueryCommand
+from ja.user.message.cancel import CancelCommand
+from typing import List, Tuple
+import copy
+
+
+class ServerProxyDummy(IUserServerProxy):
+
+    def __init__(self, ssh_config: SSHConfig):
+        self._ssh = ssh_config
+        self._jobs: List[Job] = []
+        self._counter = 0
+
+    @property
+    def jobs(self) -> List[Job]:
+        """
+        @return: the jobs on the server.
+        """
+        return self._jobs
+
+    def add_job(self, add_config: AddCommandConfig) -> Response:
+        job: Job = add_config.job
+        if job in self._jobs:
+            return Response(result_string="Job with id %s already exists" % job.uid,
+                            is_success=False)
+        job.uid = str(self._counter)
+        self._counter += 1
+        self._jobs.append(job)
+        job.status = JobStatus.QUEUED
+        return Response(result_string="Successfully added job with id: %s" % job.uid,
+                        is_success=True, uid=job.uid)
+
+    def cancel_job(self, cancel_command: CancelCommand) -> Response:
+        job_uid: str = cancel_command.uid
+        job_label: str = cancel_command.label
+        if job_uid is not None:
+            for job in self._jobs:
+                if job.uid == job_uid:
```

Comment:
Same thing as in the other file. If the job status is already cancelled, I do not think this operation should succeed.